src/Core/Localization/Currency/PatternTransformer.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Localization\Currency;
+
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+
+/**
+ * Transform a currency pattern by moving the symbol position, with or without
+ * a separation space (no-break space).
+ */
+class PatternTransformer
+{
+    const NO_BREAK_SPACE = ' ';
+    const REGULAR_SPACE = ' ';
+    const CURRENCY_SYMBOL = '¤';
+
+    const TYPE_LEFT_SYMBOL_WITH_SPACE = 'leftWithSpace';
+    const TYPE_LEFT_SYMBOL_WITHOUT_SPACE = 'leftWithoutSpace';
+    const TYPE_RIGHT_SYMBOL_WITH_SPACE = 'rightWithSpace';
+    const TYPE_RIGHT_SYMBOL_WITHOUT_SPACE = 'rightWithoutSpace';
+
+    const ALLOWED_TRANSFORMATIONS = [
+        self::TYPE_LEFT_SYMBOL_WITH_SPACE,
+        self::TYPE_LEFT_SYMBOL_WITHOUT_SPACE,
+        self::TYPE_RIGHT_SYMBOL_WITH_SPACE,
+        self::TYPE_RIGHT_SYMBOL_WITHOUT_SPACE,
+    ];
+
+    /**
+     * @var string
+     */
+    private $currencyPattern;
+
+    /**
+     * @var string
+     */
+    private $trimmedPattern;
+
+    /**
+     * @param string $currencyPattern Initial currency pattern (ex: #,##0.00¤, ¤#,##,##0.00)
+     */
+    public function __construct(string $currencyPattern)
+    {
+        $this->currencyPattern = $currencyPattern;
+
+        $trimmedCharacters = implode('', [self::CURRENCY_SYMBOL, self::NO_BREAK_SPACE, self::REGULAR_SPACE]);
```

Comment:
@jolelievre This array `[self::CURRENCY_SYMBOL, self::NO_BREAK_SPACE, self::REGULAR_SPACE]` can't be a constant ?